flycheck.el

```diff
@@ -6737,6 +6738,46 @@ See URL `http://asciidoctor.org'."
             line-end))
   :modes adoc-mode)
 
+(defun flycheck-awk-gawk-fix-message (err)
+  "Remove the repeated file-name/line from the error message of ERR."
+  (setf (flycheck-error-message err)
+        (replace-regexp-in-string
+         "^\\([ \t]*\\)\\(.*\n\\)\\1"
+         "\\2"
+         (replace-regexp-in-string
+          "\ngawk: [^ ]*:"
+          "\n"
+          (flycheck-error-message err))))
+  err)
```

Comment:
We usually prefer using `rx` over direct regexps, but I can make the change if you want :)

Reply:
rx'ed